diff --git a/vllm/v1/worker/xpu_worker.py b/vllm/v1/worker/xpu_worker.py
index 102a04f5d..a9cf693e5 100644
--- a/vllm/v1/worker/xpu_worker.py
+++ b/vllm/v1/worker/xpu_worker.py
@@ -76,6 +76,21 @@ class XPUWorker(Worker):
         os.environ["CCL_ATL_TRANSPORT"] = ENV_CCL_ATL_TRANSPORT
         os.environ["LOCAL_WORLD_SIZE"] = ENV_LOCAL_WORLD_SIZE
         os.environ["LOCAL_RANK"] = str(self.local_rank)
+        if os.getenv("VLLM_XPU_SET_CCL_LOCAL_RANK", "0") == "1":
+            tp_pp_world_size = (
+                self.parallel_config.pipeline_parallel_size
+                * self.parallel_config.tensor_parallel_size
+            )
+            dp_local_rank = self.parallel_config.data_parallel_rank_local
+            if dp_local_rank is None:
+                dp_local_rank = self.parallel_config.data_parallel_index
+            ccl_local_rank = dp_local_rank * tp_pp_world_size + self.local_rank
+            ccl_local_size = (
+                self.parallel_config.data_parallel_size_local * tp_pp_world_size
+            )
+            os.environ.setdefault("CCL_PROCESS_LAUNCHER", "none")
+            os.environ["CCL_LOCAL_RANK"] = str(ccl_local_rank)
+            os.environ["CCL_LOCAL_SIZE"] = str(ccl_local_size)
 
         init_worker_distributed_environment(
             self.vllm_config,
